eZ/Publish/Core/MVC/Symfony/Controller/Content/QueryController.php

```diff
@@ -81,4 +87,51 @@ private function runQuery(ContentView $view, $method)
         );
         $view->addParameters([$view->getParameter('query')['assign_results_to'] => $searchResults]);
     }
+
+
+    /**
+     * @param ContentView $view
+     * @param Request $request
+     * @return ContentView
+     */
+    public function pagingQueryAction(ContentView $view, Request $request)
+    {
+        $this->runPagingQuery($view, $request);
+
+        return $view;
+    }
+
+    /**
+     * @param ContentView $view
+     * @param Request $request
+     */
+    private function runPagingQuery(ContentView $view, Request $request)
+    {
+        $pager = new Pagerfanta(
+            $this->getAdapter($this->contentViewQueryTypeMapper->map($view)
+            )
+        );
+
+        $pager->setMaxPerPage( $view->getParameter('query')['limit'] );
+        $page =  $request->get('page', 1);
```

Comment:
In case of several pagers in the future (once several queries are allowed), this should perhaps be configurable by query params as well, with default being page. Or does that complicate something else here?

Reply:
In the last commit I added a parameter that could help set a specific page name for each query.